pandas/tests/frame/test_indexing.py

```diff
@@ -1137,40 +1137,40 @@ def test_ix_align(self):
         df = df_orig.copy()
 
         with catch_warnings(record=True):
```

Comment:
I think we should change these to assert_produces_warning(FutureWarning) as its more consistent with our style (could be a followup too)

Reply:
i'm not so sure.. shouldn't we only require `assert_produces_warning` for a test that specifically tests that the warning is raised (preferably with a message check) and use `@pytest.mark.filterwarnings` for all the tests where it's encountered?

Reply:
we are inconsistent in the code base for sure. but I think we *mostly* use ``assert_produces_warning``, meaning if we suddently took out the warning (or changed it), these *should* fail. checking the test of the message is ok maybe in a single case (otherwise this is overkill)

Reply:
yeah... i think we do agree, i.e. that `with catch_warnings` should not be used and replaced with either `assert_produces_warnings` or `@pytest.mark.filterwarnings`. maybe a different viewpoint on which and how often each are used?

Reply:
Whatever the discussion / decision what is best here (for new code), I don't think it is worth updating code that we won't touch until we remove it in some months / next year.

On the actual discussion, I think indeed rather the filterwarnings with pytest is the way to go for those nowadays (certainly if you need to apply it to a lot of existing cases). Of course in addition to a few tests with assert_produces_warning to actually test the warning.

Reply:
> Whatever the discussion / decision what is best here (for new code), I don't think it is worth updating code that we won't touch until we remove it in some months / next year.

agreed. as @jreback said, could be a follow-up. I don't mind looking into that.

Reply:
>  could be a follow-up. I don't mind looking into that.

You are free to spend your time how you want of course ;-) but I personally don't think it is worth the time. We will be removing this code shortly anyhow. There are plenty of other testing related things that are more useful IMO.

And wanted to suggest: like documenting our standard what to do in those case, but it is actually already quite good documented: http://pandas-docs.github.io/pandas-docs-travis/development/contributing.html#testing-warnings (and so exactly how you suggested it above)

Reply:
i'll need to check, but from memory I think `.ix` is used in a few tests to generate the expected result for testing the other indexers.

So maybe the first step would be to eliminate these, (would need to be done before .ix deprecation anyway). This would allow the removal of some catch_warnings.

I would guess that in theory it should be possible to remove all `.ix` occurrences from test files outside the test_ix module.

Reply:
yep that would be great; best to just use iloc for the expected